### PR TITLE
Conveyors Code + Sprites Overhaul

### PR DESCRIPTION
**Pull request by GoldenAlpharex**
## About The Pull Request
So, I wanted to fix something with conveyor belts acting weird in corners, and it ended up being a bit of a bigger undertaking as I've ended up refactoring some conveyor icon behavior for corners, as well as documenting all of this ten-years-old code, improving it where I could and especially improving variable names.

I was a little sad from how little information you would get from conveyor belts in-game, so I added some more information in the description, which should hopefully help conveying (tee hee) how conveyors work in-game.

I also decided that, to make it worth it, a sprite overhaul would be in order! Thus, I decided to port [CEV Eris' conveyor and conveyor switches sprites, made by SpadesSlick](https://github.com/discordia-space/CEV-Eris/pull/3893) ~(Apparently it was them and not Tau Ceti)~ Apparently it was made by WalterJe, but I couldn't find their original PR, I pulled the assets from Eris, but they're the spriter that did them. Regardless, they're very nice and, unlike the current conveyor belts, tell you at a glance where they're transporting items. The switches are also looking less like twinks and more like actual levers, as they were always designed to be.

*But what do they look like, Golden?*

Glad you asked. Here's a gif of what they actually look like in-game:
New conveyor sprites

- [x] The only thing left on this PR for it to be ready is to get some ~new item sprites for the belts, as well as some~(found them) new broken sprites, which I wasn't able to find on Eris, but hopefully I'll find someone wonderful to help me with those :)

## Why It's Good For The Game
Updating some old-ass code to be more fit for today's standards, more accessibility for conveyor belts and understanding how it works without code-diving every time, behavior in-game that's coherent to what's expected (looking at you, conveyor corners), better documentation for the next person that'll have to touch this mess, better sprites... That's all a win in my book.

## Changelog

:cl: GoldenAlpharex, sprites from WalterJe
qol: Conveyors will now also convey how to interact with them with tools upon being examined.
imageadd: Updated the sprites of the conveyor belts and the conveyor switches, ported and adapted from Tau Ceti!
refactor: Brought up to today's standards conveyor codes, alongside properly documenting it. Also changed how conveyor corners are handled to make them both more intuitive for mappers and for people in-game.
/:cl:

### CODE REVIEW

**Comment by Toastgoats** [comment]
Very nice, it'd probably be hell to add but some shading would be nice to have in the future.

**Comment by datorangebottle** [comment]
I love this.

**Comment by Sealed101** [comment]
seems like tau ceti ported these over from cev eris according to the og pr description. 
lovely sprites

**Comment by EricZilla** [comment]
Love those belts

File: code/modules/recycling/conveyor.dm [hunk 1]
```diff
@@ -137,61 +175,86 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	var/list/items = locturf.contents - src
 	if(!LAZYLEN(items))//Dont do anything at all if theres nothing there but the conveyor
 		return
+
```
Comment (SuperNovaa41) [inline code/modules/recycling/conveyor.dm:178]: why did you add so many spaces into places you didn't touch the code

Comment (GoldenAlpharex) [reply code/modules/recycling/conveyor.dm:178]: Because it makes it easier to read, the less dense the code is, the easier it is to understand the logic of it, it's less mental workload, etc.

File: code/modules/recycling/conveyor.dm [hunk 2]
```diff
@@ -99,26 +127,36 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 		if(SOUTHWEST)
 			forwards = WEST
 			backwards = NORTH
-	if(verted == -1)
+
+	if(inverted)
+		var/temp = forwards
+		forwards = backwards
+		backwards = temp
+	if(flipped) // We need to do this this way to ensure good functionality on corner belts.
```
Comment (optimumtact) [inline code/modules/recycling/conveyor.dm:135]: expand your thesis here, why does doing it this way work better for corners?

Comment (optimumtact) [reply code/modules/recycling/conveyor.dm:135]: i.e the comment, not reply to my review

Comment (GoldenAlpharex) [reply code/modules/recycling/conveyor.dm:135]: Right, right.

**Review by Imaginos16 — APPROVED** [review]
On the sprite side of things, this is undeniably an improvement! Regular maintainers have to approve or merge the code side of things however.

**Comment by Kurgis** [comment]
> GoldenAlpharex, sprites from SpadesSlick over at CEV Eris
Incorrect attribution, sprites were may be WatlerJe.

**Comment by GoldenAlpharex** [comment]
> > GoldenAlpharex, sprites from SpadesSlick over at CEV Eris
> 
> Incorrect attribution, sprites were made by WatlerJe.

Oh, is that true? I actually haven't been able to find who made them, I pointed to the one PR I saw which seemed to be the first one? It's hard to tell when dealing with Russian codebases. Got a link to where they posted it first, or, mentioned it was by them? I totally want to credit the right person, I just don't want to keep playing ping-pong ;)

**Comment by Kurgis** [comment]
Trouble is most of these things are done in Discord channels so it's not like there's accessible history for them, SpadesSlick was the Project Lead for Eris, they were just pushing some of Watler's work to the repo. This is all, however, second-hand information, and of course Discord is a poor platform to share game assets in an organized manner. Even so, all of Eris' assets are pretty much stored as embedded Discord files (sometimes in random channels), of course this makes it pretty much impossible due to the shoddy search function they have to really find anything. Spades is currently on hiatus, so you'd need to join the TauCeti Discord and DM WatlerJe if you want to directly confirm this is their work.

**Comment by Kurgis** [comment]
> I just don't want to keep playing ping-pong

This is also a reason many people don't specifically credit a name, instead using something vague like "tgstation" or "from X repo", it currently takes more effort than it reasonably should often times. Most Eris sprite PRs usually just list "tgstation" for the same reason.

**Comment by cacogen** [comment]
The old sprites looked more realistic but I've always wanted arrows for finding conveyors that are facing the wrong way

**Comment by NanoCats** [comment]
I don't like the new levers, I think they clash a bit, too many pixels, too much fading, but I don't care that much and I'll happily take them if it means I can actually tell which way conveyers are oriented without fookin putting something on them and trial/erroring.

**Comment by LemonInTheDark** [comment]
image
I am exempting the QOL change here (Examine description) because it would be stupid not to. 
QOL prs are still against the freeze, because I don't trust you motherfuckers not to just call all your minor features QOL

**Comment by GoldenAlpharex** [comment]
> I don't like the new levers, I think they clash a bit, too many pixels, too much fading, but I don't care that much and I'll happily take them if it means I can actually tell which way conveyers are oriented without fookin putting something on them and trial/erroring.

If that comes to be too big of an issue going forward, we can always change them, I think they're simply much easier to interact with, trying to click on two pixels can be really annoying, so much so I always just went for the base of the lever instead.

**Comment by Likteer** [comment]
You have Likteer's endorsment.
Somewhat random, but @GoldenAlpharex - do conveyor belts work fully independently from regular movement, or can different floors affect their speed?

File: code/modules/recycling/conveyor.dm [hunk 3]
```diff
@@ -10,21 +17,37 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
 	processing_flags = START_PROCESSING_MANUALLY
 	subsystem_type = /datum/controller/subsystem/processing/conveyors
-	var/operating = 0 // 1 if running forward, -1 if backwards, 0 if off
-	var/operable = 1 // true if can operate (no broken segments in this belt run)
-	var/forwards // this is the default (forward) direction, set by the map dir
-	var/backwards // hopefully self-explanatory
-	var/movedir // the actual direction to move stuff in
-	var/id = "" // the control ID - must match controller ID
-	var/verted = 1 // Inverts the direction the conveyor belt moves.
+	/// The current operating status. CONVEYOR_OFF (0) if off, CONVEYOR_FORWARD (1) if running forward, CONVEYOR_BACKWARDS (-1) if backwards.
```
Comment (LemonInTheDark) [inline code/modules/recycling/conveyor.dm:20]: why include the values?

File: code/modules/recycling/conveyor.dm [hunk 4]
```diff
@@ -10,21 +17,37 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
+	var/forwards
+	/// The opposite of forwards. It's set in a special var for corner belts, which aren't using the opposite direction when in reverse.
+	var/backwards
+	/// The actual direction to move stuff in.
+	var/movedir
+	/// The control ID - must match at least one conveyor switch's ID to be useful.
+	var/id = ""
+	/// Inverts the direction the conveyor belt moves when true.
+	var/inverted = FALSE
+	/// Is the conveyor's belt flipped? Useful mostly for conveyor belt corners. DIFFERENT FROM INVERTED FOR CORNERS!
```
Comment (LemonInTheDark) [inline code/modules/recycling/conveyor.dm:32]: How is it different then inverted?

File: code/modules/recycling/conveyor.dm [hunk 5]
```diff
@@ -99,99 +125,135 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
+
+	if(inverted)
+		var/temp = forwards
+		forwards = backwards
+		backwards = temp
+	// We need to do this this way to ensure good functionality on corner belts.
+	// Basically, this allows the conveyor belts that used a flipped belt sprite to
+	// still convey items in the direction of their arrows. It's different from inverted,
+	// so they need to be ran separately.
+	if(flipped)
```
Comment (LemonInTheDark) [inline code/modules/recycling/conveyor.dm:137]: why not just flipped || inverted?

File: code/modules/recycling/conveyor.dm [hunk 6]
```diff
@@ -99,99 +125,135 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
 
 /obj/machinery/conveyor/update_icon_state()
-	icon_state = "[base_icon_state][(machine_stat & BROKEN) ? "-broken" : (operating * verted)]"
+	icon_state = "[base_icon_state][inverted ? -operating : operating ][flipped ? "-flipped" : ""]"
 	return ..()
 
+/**
+ * Proc to update the conveyor depending on if it's without power (there used to be more
+ * checks, but they were broken and thus removed).
+ *
```
Comment (LemonInTheDark) [inline code/modules/recycling/conveyor.dm:154]: Shouldn't document what doesn't exist anymore

File: code/modules/recycling/conveyor.dm [hunk 7]
```diff
@@ -99,99 +125,135 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
 	if(length(items) > MAX_CONVEYOR_ITEMS_MOVE)
-		affecting = items.Copy(1, MAX_CONVEYOR_ITEMS_MOVE + 1)//Lists start at 1 lol
+		affecting = items.Copy(1, MAX_CONVEYOR_ITEMS_MOVE + 1) // Lists start at 1 lol.
 	else
 		affecting = items
+
 	conveying = TRUE
 
-	addtimer(CALLBACK(src, .proc/convey, affecting), 1)//Movement effect
+	addtimer(CALLBACK(src, .proc/convey, affecting), 1) // Movement effect.
```
Comment (LemonInTheDark) [inline code/modules/recycling/conveyor.dm:190]: Comments don't nessesarially need ., don't feel like you need to add em everywhere.
The space after is also just a style choice, but since you're redoing the whole file s ok

File: code/modules/recycling/conveyor.dm [hunk 8]
```diff
@@ -246,17 +278,22 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	base_icon_state = "switch"
 	processing_flags = START_PROCESSING_MANUALLY
 
-	var/position = 0 // 0 off, -1 reverse, 1 forward
-	var/last_pos = -1 // last direction setting
-	var/oneway = FALSE // if the switch only operates the conveyor belts in a single direction.
-	var/invert_icon = FALSE // If the level points the opposite direction when it's turned on.
-
-	var/id = "" // must match conveyor IDs to control them
+	/// The current direction of the switch. CONVEYOR_OFF (0) if off, CONVEYOR_FORWARD (1) if tilted forward, CONVEYOR_BACKWARDS (-1) if tilted backwards..
```
Comment (LemonInTheDark) [inline code/modules/recycling/conveyor.dm:281]: ditto again

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/recycling/conveyor.dm (hunk 4 above)
Comment (GoldenAlpharex) [reply code/modules/recycling/conveyor.dm:32]: True, I should be clearer.

File: code/modules/recycling/conveyor.dm (hunk 5 above)
Comment (GoldenAlpharex) [reply code/modules/recycling/conveyor.dm:137]: Because they both need to be ran separately.

File: code/modules/recycling/conveyor.dm (hunk 6 above)
Comment (GoldenAlpharex) [reply code/modules/recycling/conveyor.dm:154]: True.

File: code/modules/recycling/conveyor.dm (hunk 7 above)
Comment (GoldenAlpharex) [reply code/modules/recycling/conveyor.dm:190]: > Comments don't nessesarially need ., don't feel like you need to add em everywhere. The space after is also just a style choice, but since you're redoing the whole file s ok

Oh, yeah, I know. I just thought I'd go ahead and do that, it's just personal preference at this point, I'm not too worried about it either way. Just space is just better for visibility in my opinion. The period, yeah, that doesn't honestly matter too much.

File: code/modules/recycling/conveyor.dm (hunk 3 above)
Comment (GoldenAlpharex) [reply code/modules/recycling/conveyor.dm]: Removed, I reworded it in a way that should hopefully be clearer.

File: code/modules/recycling/conveyor.dm (hunk 5 above)
Comment (LemonInTheDark) [reply code/modules/recycling/conveyor.dm]: oh they both need to apply

File: code/modules/recycling/conveyor.dm [hunk 9]
```diff
@@ -99,99 +124,134 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
-			if(!(machine_stat & BROKEN))
-				var/obj/item/stack/conveyor/C = new /obj/item/stack/conveyor(loc, 1, TRUE, null, null, id)
-				transfer_fingerprints_to(C)
-			to_chat(user, span_notice("You remove the conveyor belt."))
+/obj/machinery/conveyor/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(attacking_item.tool_behaviour == TOOL_CROWBAR)
+		user.visible_message(span_notice("[user] struggles to pry up [src] with [attacking_item]."), \
+		span_notice("You struggle to pry up [src] with [attacking_item]."))
+
+		if(attacking_item.use_tool(src, user, 40, volume = 40))
```
Comment (LemonInTheDark) [inline code/modules/recycling/conveyor.dm:223]: early return?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
One minor thing, then we're good
